compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -6834,6 +6837,8 @@ private BType checkInvocationArgs(BLangInvocation iExpr, List<BType> paramTypes,
                     dlog.error(arg.pos, DiagnosticErrorCode.UNDEFINED_PARAMETER, argName);
                     break;
                 }
+                populateIncludedRecordParams(varSym, incRecords);
+                namedArgs.add(arg);
```

Comment:
Without creating a separate list for named arguments can't we refactor the overall logic as follows,
- When visiting a named arg we can check whether the symbol of that argument in the `incRecordParams` list or not

Reply:
Changed the logic, please check